crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs

```diff
@@ -0,0 +1,114 @@
+use ruff_diagnostics::AlwaysFixableViolation;
+use ruff_diagnostics::{Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::whitespace::trailing_comment_start_offset;
+use ruff_python_ast::Stmt;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::fix;
+
+/// ## What it does
+/// Checks for unnecessary `pass` statements in functions, classes, and other
+/// blocks.
+///
+/// In [preview], this rule also checks for unnecessary ellipsis (`...`)
+/// expressions.
+///
+/// ## Why is this bad?
+/// In Python, the `pass` statement and ellipsis (`...`) expression serve as
+/// placeholders, allowing for syntactically correct empty code blocks. The
+/// primary purpose of these nodes is to avoid syntax errors in situations
+/// where a statement or expression is syntactically required, but no code
+/// needs to be executed.
+///
+/// If a `pass` or ellipsis is present in a code block that includes at least
+/// one other statement (even, e.g., a docstring), it is unnecessary and should
+/// be removed.
+///
+/// ## Example
+/// ```python
+/// def func():
+///     """Placeholder docstring."""
+///     pass
+/// ```
```

Comment:
Can we add an example with `...`, probably stating that this is only applicable in preview mode?